components/BottomNavigation/tests/snapshot/MDCBottomNavigationBarControllerSnapshotTests.m

```diff
@@ -0,0 +1,230 @@
+@property(nonatomic, readonly, nonnull) UICollectionViewFlowLayout *flowLayout;
+@end
+
+@implementation MDCBottomNavigationBarControllerSnapshotTestScrollableChildViewController
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  [self.collectionView registerClass:[UICollectionViewCell class]
+          forCellWithReuseIdentifier:@"cell"];
+  self.collectionView.dataSource = self;
+  if (@available(iOS 11.0, *)) {
+    self.collectionView.insetsLayoutMarginsFromSafeArea = YES;
+    self.collectionView.contentInsetAdjustmentBehavior = UIScrollViewContentInsetAdjustmentAlways;
+  }
+}
+
+- (UICollectionViewFlowLayout *)flowLayout {
+  return (UICollectionViewFlowLayout *)self.collectionViewLayout;
+}
+
+- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
+  return 1;
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView
+     numberOfItemsInSection:(NSInteger)section {
+  return kNumberOfCellsInScrollableChild;
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
+                  cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+  CGFloat hue = (CGFloat)indexPath.row / kNumberOfCellsInScrollableChild;
+  UICollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"cell"
+                                                                         forIndexPath:indexPath];
+  cell.backgroundColor = [UIColor colorWithHue:hue saturation:1 brightness:1 alpha:1];
+  return cell;
+}
+
+@end
+
+/** Snapshot tests for @c MDCBottomNavigationBarController. */
+@interface MDCBottomNavigationBarControllerSnapshotTests : MDCSnapshotTestCase
+
+/** The view controller being tested. */
+@property(nonatomic, strong) MDCBottomNavigationBarController *navBarController;
+
+/** A child view controller that has scrollable content. */
+@property(nonatomic, strong)
+    MDCBottomNavigationBarControllerSnapshotTestScrollableChildViewController *scrollableChildVC;
+
+/** A child view controller with non-scrolling content. */
+@property(nonatomic, strong)
+    MDCBottomNavigationBarControllerSnapshotTestFixedContentChildViewController
+        *fixedContentChildVC;
+
+@end
+
+@implementation MDCBottomNavigationBarControllerSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  //  self.recordMode = YES;
+
+  self.navBarController = [[MDCBottomNavigationBarController alloc] init];
+  self.navBarController.navigationBar.barTintColor =
+      [UIColor.redColor colorWithAlphaComponent:(CGFloat)0.5];
+  self.navBarController.view.backgroundColor = UIColor.whiteColor;
+  self.navBarController.view.bounds = CGRectMake(0, 0, 240, 360);
+
+  self.fixedContentChildVC =
+      [[MDCBottomNavigationBarControllerSnapshotTestFixedContentChildViewController alloc] init];
+  self.fixedContentChildVC.view.backgroundColor = UIColor.blueColor;
```

Comment:
Can we make this either white or black? It would help me to confirm I'm looking at the last square.

Reply:
Done.